workflows/ncn/hooks/before-all/move-critical-singleton-pods.yaml

```diff
@@ -52,9 +52,10 @@ spec:
 
     echo
 
-    pods=( "nexus" "ipxe" "kea" "cray-cfs-api-db" )
-    for pod in ${pods[@]};do
-      podName=$(kubectl get pods --all-namespaces | awk '{print $2}' | grep $pod)
+    containerNames=( "nexus" "ipxe" "kea" "cray-cfs-api-db" )
+    for containerName in ${containerNames[@]};do
+      # get pod name by container name
+      podName=$(kubectl get pods -A -o=jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.status.phase}{"\t"}{range .spec.containers[*]}{.name}{"\t"}{end}{"\n"}{end}' | grep -E "^[^[:space:]]+[[:space:]]+Running[[:space:]]+.*${containerName}.*$" | cut -f1)
```

Comment:
This is too complicated and hard to follow.  It is common kubectl practice to use labels for this specific situation rather than grep/awk/cut.

For example:
```
ncn-m001:~ # kubectl get pods -A -l app.kubernetes.io/name=cray-cfs-api-db -o custom-columns=PodName:.metadata.name --no-headers=true
cray-cfs-api-db-5fddcf7864-m6qsk
```

Reply:
The approach above will still catch cases where a pod is unexpected created that has a container with "nexus" or "ipxe" or "kea" as a substring of the container name.

Reply:
https://kubernetes.io/docs/reference/labels-annotations-taints/

Reply:
while i agree we should use label but i'd disagree that we should use any of our existing labels. We should just have `critical-pod:true` so i can also remove line 55. any pod requires a move during upgrade should declare that.

Reply:
I'm not following why we shouldn't use our existing labels.   For this set of pods we have labels that are specific only to those pods.   I'm okay if you want to add a {{critical-pod:true}} label at some point to be able to get all of the critical pods with one label.   But that is going to take some time to get those labels added and we won't be able to rely on that for older pods.   Using the labels that are there now will be much better than doing the greps that are being done right now.